stream-plan-manager/plan-manager-core/src/main/java/com/backbase/streams/tailoredvalue/plan/PlansSaga.java

```diff
@@ -0,0 +1,88 @@
+package com.backbase.streams.tailoredvalue.plan;
+
+import com.backbase.stream.worker.StreamTaskExecutor;
+import com.backbase.stream.worker.exception.StreamTaskException;
+import com.backbase.streams.tailoredvalue.configuration.PlansProperties;
+import com.backbase.tailoredvalue.planmanager.service.api.v0.PlansApi;
+import com.backbase.tailoredvalue.planmanager.service.api.v0.UserPlansApi;
+import com.backbase.tailoredvalue.planmanager.service.api.v0.model.PlansGetResponseBody;
+import com.backbase.tailoredvalue.planmanager.service.api.v0.model.UserPlanUpdateRequestBody;
+import jakarta.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+@Slf4j
+@RequiredArgsConstructor
+public class PlansSaga implements StreamTaskExecutor<PlansTask> {
```

Comment:
Without any orchestration and no rollback, there is not much Saga here.

Reply:
I intitially followed some other Sagas like audiences - https://github.com/Backbase/stream-services/tree/master/stream-audiences. 
Refactored just to have a service calls like product-catalog ot payments